### PR TITLE
convolution

### PR DESCRIPTION
**Pull request by eelfroth**
ronin-convolve

### CODE REVIEW

File: desktop/sources/scripts/library.js [hunk 1]
```diff
@@ -287,29 +287,62 @@ function Library (ronin) {
...
     }
-    ronin.surface.context.putImageData(img, 0, 0)
+    ronin.surface.context.putImageData(img, rect.x, rect.y)
     return rect
   }
 
   this.saturation = (pixel, q = 1) => {
-    const color = 0.2126 * pixel.r + 0.7152 * pixel.g + 0.0722 * pixel.b
-    return [(color * (1 - q)) + (pixel.r * q), (color * (1 - q)) + (pixel.g * q), (color * (1 - q)) + (pixel.b * q), pixel.a]
+    const color = 0.2126 * pixel.r + 0.7152 * pixel.g + 0.0722 * pixel.b * (1 - q)
```
Comment (lctrt) [inline desktop/sources/scripts/library.js]: not sure about the saturation change, this * (1 - q) multiplication is only on the 3rd color

File: desktop/sources/scripts/library.js [hunk 2]
```diff
@@ -287,29 +287,62 @@ function Library (ronin) {
   // Pixels
 
   this.pixels = (rect, fn, q) => {
-    const img = ronin.surface.context.getImageData(0, 0, rect.w, rect.h)
+    const img = ronin.surface.context.getImageData(rect.x, rect.y, rect.w, rect.h)
```
Comment (lctrt) [inline desktop/sources/scripts/library.js]: ⭐️

**Comment by neauoire** [comment]
Niiiiiice! Could we also have an index for strenght of the filter? 

`(convolution blur 0.4)`

**Comment by neauoire** [comment]
Actually i think I can adapt it :)

File: desktop/sources/scripts/library.js (hunk 1 above)
Comment (neauoire) [reply desktop/sources/scripts/library.js]: I'll correct

**Comment by neauoire** [comment]
That works amazingly well.

File: desktop/sources/scripts/library.js (hunk 1 above)
Comment (eelfroth) [reply desktop/sources/scripts/library.js]: oh oops, sorry about that ^_^

**Pull request merged**